pkg/datapath/linux/node.go

```diff
@@ -575,6 +576,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 			// a gateway. Send arping to the gw IP addr instead of newNode IP addr.
 			// NOTE: we currently don't handle multipath, so only one gw can be used.
 			copy(nextHopIPv4, route.Gw)
+			copy(srcIP, route.Src)
```

Comment:
We don't know if `route.Src` as size 4. We need to either make `srcIP` with size `net.IPv6Len` or do:
```
routeSrcIPv4 := route.Src.To4()
```